Update airbrake version to be compatible with latest rails and ruby

cintamani opened the pull request:
From: https://eaflood.atlassian.net/browse/RUBY-1060

cintamani committed:
Update airbrake version to be compatible with latest rails and ruby

From: https://eaflood.atlassian.net/browse/RUBY-1060

cintamani committed:
Update travis configs

cintamani commented on .travis.yml:
```diff
@@ -16,7 +16,7 @@ git:
 before_install:
   - export TZ=UTC
-  - gem install -v 1.17.3 bundler --no-rdoc --no-ri
+  - gem install bundler --no-rdoc --no-ri
```
```suggestion
  - gem install bundler --no-document
```

cintamani committed:
Update .travis.yml

cintamani commented:
Seems like the newest version of Airbrake does not have the close need from the failures on travis. Leaving it here to gather the rest of the team opinion on this.

Cruikshanks commented:
I think if we can recreate throwing an error during a rake task and confirming it gets logged in Errbit, then we should be good to stop using that aspect of the gem. This was the original reason for including it (and creating this gem); we were missing notifications in Errbit from failed rake tasks because Airbrake wasn't being given a chance to close.

Cruikshanks commented:
I opened another PR based on this one to look at the issue. Go here for details https://github.com/DEFRA/defra-ruby-alert/pull/9#issuecomment-638487028. But after some playing around I'm happy that we should be removing our `at_exit` hook logic and that this will fix our issue with this gem.

Unfortunately, I ran out of time this evening to actually make the change to my PR. But will do it first thing tomorrow!

Cruikshanks committed:
Remove airbrake close hook from gem (#9)

https://eaflood.atlassian.net/browse/RUBY-1060

This is part of work to get our services using the latest version of Rails and Ruby (6 and 2.7.1).

Previously this gem was using version 5.8.1 and 5.3.0 of the Airbrake gem (pre-ruby upgrade for our services). Both versions define their own `at_exit` hook but don't call `Airbrake.close` in it.

We know not calling `Airbrake.close` is an issue for things like rake tasks, where if an exception is thrown the app is likely to close before Airbrake can send the notification. This is why we implement our own `at_exit hook.

This issue was highlighted in [Airbrake issue 695](https://github.com/airbrake/airbrake/issues/695), which they then went and fixed in the ***very next version*** (6.0.0)! They did this by adding a call to `Airbrake.close()` in their own `at_exit` hook.

This not only makes our `at_exit` no longer needed, it actually causes an error. If you call Airbrake.close() more than once it will throw `raise Airbrake::Error, 'this thread pool is closed already`.

This is why our tests now fail and why any app trying to use a version of this gem with a version of Airbrake greater than 5.8.1 will throw an error on exit. 😁🎉

So this change covers removing the `at_exit` hook and the call to `Airbrake.close()` from our gem.
